.github/CODING_GUIDELINES.md

```diff
@@ -252,3 +253,109 @@ public class ThisIsASampleClassTest()
     }
 }
 ```
+
+# Commit Message Guidelines
+We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.
+
+**Commit Message Format**
+
+Each commit message consists of a header, a body (optional) and a footer (optional). The header has a special format that includes a type, a scope and a subject:
+
+```java
+
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+
+```
+The header is mandatory and the body and footer are optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier to read on GitHub as well as in various git tools.
+
+Footer should contain a closing reference to an issue if any.
+
+**Revert**
+
+If the commit reverts a previous commit, it should begin with revert:, followed by the header of the reverted commit. In the body it should say: This reverts commit <hash>., where the hash is the SHA of the commit being reverted.
+
+**Squash**
+
+Split your commits on the basis of the package you are providing code to. 
+
+Squash all your commits per scope (i.e package). It is also important to rebase your commits on master.
+
+**Type**
+
+Must be one of the following:
+
+* build: Changes that affect the build system (eg POM changes)
+* docs: Documentation only changes (eg ReadMe)
+* feat: A new feature
+* fix: A bug fix
+* perf: A code change that improves performance
+* refactor: A code change that neither fixes a bug nor adds a feature
+* style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* test: Adding missing tests or correcting existing tests
+
+**Scope**
+
+The scope should be the name of the maven package affected as perceived by person reading changelog generated from commit messages.
+
+The following is the list of supported scopes:
+
+* iot-dev
+* iot-dev-samples
+* iot-service
+* iot-service-samples
+* deps
+* prov-dev
+* prov-service
+* prov-samples
+* security-provider
+* tpm-provider
+* tpm-emulator
+* x509-provider
+* dice-emulator
+* dice-provider
+
+**Subject**
+
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* no dot (.) at the end
+
+**Body**
+
+Just as in the subject, use the imperative, present tense: 
+
+"change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
```

Comment:
Do we need to state that? Can't we point to a link with that common practice?

Reply:
Will do that